log/log.go

```diff
@@ -39,10 +41,63 @@ func (f levelFlag) Set(level string) error {
 	return nil
 }
 
+func setFormatterSylog(appname, local string) error {
+	if appname != "" && local != "" {
+		return nil
+	}
+
+	fmter, err := newSyslogger(appname, local, origLogger.Formatter)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error creating syslog formatter: %v\n", err)
+		origLogger.Errorf("can't connect logger to syslogs: %v", err)
+		return err
+	}
+	origLogger.Formatter = fmter
+	return nil
+}
+
+func setJsonFormatter() {
+	origLogger.Formatter = &logrus.JSONFormatter{}
+}
+
+type formatLog struct{}
+
+// String implements flag.Value.
+func (f formatLog) String() string {
+	return origLogger.Level.String()
+}
+
+// Set implements flag.Value.
+func (f formatLog) Set(format string) error {
+	u, err := url.Parse(format)
+	if err != nil {
+		return err
+	}
+	if u.Scheme != "logger" {
+		return fmt.Errorf("invalid scheme %s", u.Scheme)
+	}
+	jsonq := u.Query().Get("json")
+	if jsonq == "true" {
+		setJsonFormatter()
+	}
+
+	switch u.Opaque {
+	case "syslog":
+		appname := u.Query().Get("appname")
+		local := u.Query().Get("local")
+		return setFormatterSylog(appname, local)
+	case "stdout":
+	default:
+		return fmt.Errorf("unsupported logger", u.Opaque)
```

Comment:
Ah sorry, I forgot there isn't a `fmt.Error` method like the `log.Error` one. Please add the missing ` %s` to print the logger value (`go vet` should also mark this problem).